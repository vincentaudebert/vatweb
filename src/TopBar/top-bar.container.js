import { connect } from 'react-redux';
import TopBar from './top-bar';

const mapStateToProps = (state, ownProps) => {
  return {
    openFiles: state.openFiles,
  };
};

const mapDispatchToProps = (dispatch, ownProps) => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TopBar);
