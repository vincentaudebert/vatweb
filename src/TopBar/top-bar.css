.top-bar {
  background-color: #3c3c3c;
  font-size: 0.875rem; }

.top-bar__btn-group {
  margin-left: 0.75rem !important; }

.top-bar__btn {
  height: 0.875rem;
  width: 0.875rem;
  border-radius: 50%;
  margin-left: 0.75rem;
  border: 0;
  font-size: 0.625rem;
  padding-left: 0.0625rem;
  padding-top: 0rem; }
  .top-bar__btn:first-child {
    margin-left: 0; }
  .top-bar__btn:focus {
    outline: 0; }
  .top-bar__btn:hover {
    color: #1e1e1e; }

.top-bar__btn--close {
  color: #ff5c5c;
  background-color: #ff5c5c;
  border: 1px solid #ff4343; }
  .top-bar__btn--close:active {
    background: #c14645;
    border: 1px solid #b13c3b; }

.top-bar__btn--minimize {
  color: #ffbd4c;
  background-color: #ffbd4c;
  border: 1px solid #ffb433; }
  .top-bar__btn--minimize:active {
    background: #c08e38;
    border: 1px solid #ac7f32; }

.top-bar__btn--expand {
  color: #00ca56;
  background-color: #00ca56;
  border: 1px solid #00b14b; }
  .top-bar__btn--expand:active {
    background: #029740;
    border: 1px solid #027e35; }
