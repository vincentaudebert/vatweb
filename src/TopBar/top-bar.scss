@import '../css/variables';
@import '../css/mixins';

.top-bar {
  background-color: $color-medium-grey;
  font-size: rem(14px);

  .top-bar__item,
  .top-bar__filename {
    margin-top: rem(4px);
    margin-bottom: rem(4px);
  }
}

.top-bar__filename {
  position: absolute;
  width: calc(
    100% - 132px
  ); // left menu (close, minimize, etc) is 66px so we times 2 to center it
  margin-left: auto;
  margin-right: auto;
}

.top-bar__btn {
  height: rem(14px);
  width: rem(14px);
  border-radius: 50%;
  margin-left: $spacing;
  border: 0;
  font-size: rem(10px);
  padding-left: rem(1px);
  padding-top: rem(0px);

  &:first-child {
    margin-left: 0;
  }

  &:focus {
    outline: 0;
  }

  &:hover {
    color: $color-dark-grey;
  }
}

.top-bar__btn--close {
  color: $color-red;
  background-color: $color-red;
  border: 1px solid $color-darker-red;

  &:active {
    background: $color-hover-red;
    border: 1px solid $color-darker-hover-red;
  }
}

.top-bar__btn--minimize {
  color: $color-yellow;
  background-color: $color-yellow;
  border: 1px solid $color-darker-yellow;

  &:active {
    background: $color-hover-yellow;
    border: 1px solid $color-darker-hover-yellow;
  }
}

.top-bar__btn--expand {
  color: $color-green;
  background-color: $color-green;
  border: 1px solid $color-darker-green;

  &:active {
    background: $color-hover-green;
    border: 1px solid $color-darker-hover-green;
  }
}
