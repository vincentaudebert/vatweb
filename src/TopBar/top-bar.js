import React from 'react';
import './top-bar.css';
import MdClose from 'react-icons/lib/md/close';
import IoMinus from 'react-icons/lib/io/minus';
import FaExpand from 'react-icons/lib/fa/expand';
import { getActiveItem } from '../utilities';

export const TopBar = props => {
  const currentFile = getActiveItem(props.openFiles);
  const name = currentFile ? currentFile.name : undefined;

  return (
    <div className="top-bar row">
      <div className="col-4-sm left ml-spacing top-bar__item">
        <button
          type="button"
          onClick={() => props.toggleEditor(false)}
          className="top-bar__btn top-bar__btn--close"
        >
          <MdClose />
        </button>
        <button
          type="button"
          value="Reduce"
          className="top-bar__btn top-bar__btn--minimize"
        >
          <IoMinus />
        </button>
        <button
          type="button"
          value="Expand"
          className="top-bar__btn top-bar__btn--expand"
        >
          <FaExpand />
        </button>
      </div>
      <div className="col-4-sm center top-bar__item">
        {name && `${name} - `}vatweb.fr
      </div>
    </div>
  );
};

export default TopBar;
