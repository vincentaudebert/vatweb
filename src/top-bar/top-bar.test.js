import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TopBar from './top-bar';

describe('TopBar', () => {
  let currentFile;
  let toggleEditor;

  beforeEach(() => {
    toggleEditor = jest.fn();
    currentFile = {
      name: 'Test Name',
      content: 'Test Content',
      inFolder: false,
      language: 'react',
      type: 'file',
    };
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<TopBar currentFile={currentFile} toggleEditor={toggleEditor} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with undefined currentFile', () => {
    currentFile = undefined;
    const tree = renderer
      .create(<TopBar currentFile={currentFile} toggleEditor={toggleEditor} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('close button toggles editor', () => {
    const topBar = shallow(
      <TopBar currentFile={currentFile} toggleEditor={toggleEditor} />
    );
    topBar.find('button.top-bar__btn--close').simulate('click');

    expect(toggleEditor).toHaveBeenCalled();
  });
});
