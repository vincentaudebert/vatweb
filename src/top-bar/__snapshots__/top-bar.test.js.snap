// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<div
  className="top-bar row"
>
  <div
    className="d-inline-block left ml-spacing top-bar__left"
  >
    <button
      className="top-bar__btn top-bar__btn--close"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </button>
    <button
      className="top-bar__btn top-bar__btn--minimize"
      type="button"
      value="Reduce"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m5 22.5v-5h30v5h-30z"
          />
        </g>
      </svg>
    </button>
    <button
      className="top-bar__btn top-bar__btn--expand"
      type="button"
      value="Expand"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m19.9 23.6q0 0.3-0.3 0.5l-7.4 7.4 3.2 3.2q0.5 0.4 0.5 1t-0.5 1-1 0.4h-10q-0.6 0-1-0.4t-0.4-1v-10q0-0.6 0.4-1t1-0.4 1 0.4l3.2 3.2 7.5-7.4q0.2-0.2 0.5-0.2t0.5 0.2l2.5 2.6q0.3 0.2 0.3 0.5z m17.4-19.3v10q0 0.6-0.4 1t-1 0.4-1-0.4l-3.3-3.2-7.4 7.4q-0.2 0.2-0.5 0.2t-0.5-0.2l-2.5-2.6q-0.3-0.2-0.3-0.5t0.3-0.5l7.4-7.4-3.2-3.2q-0.5-0.4-0.5-1t0.5-1 1-0.4h10q0.5 0 1 0.4t0.4 1z"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="d-inline-block center top-bar__right"
  >
    Test Name - 
    vatweb.fr
  </div>
</div>
`;

exports[`TopBar renders correctly with undefined currentFile 1`] = `
<div
  className="top-bar row"
>
  <div
    className="d-inline-block left ml-spacing top-bar__left"
  >
    <button
      className="top-bar__btn top-bar__btn--close"
      onClick={[Function]}
      type="button"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m31.6 10.7l-9.3 9.3 9.3 9.3-2.3 2.3-9.3-9.3-9.3 9.3-2.3-2.3 9.3-9.3-9.3-9.3 2.3-2.3 9.3 9.3 9.3-9.3z"
          />
        </g>
      </svg>
    </button>
    <button
      className="top-bar__btn top-bar__btn--minimize"
      type="button"
      value="Reduce"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m5 22.5v-5h30v5h-30z"
          />
        </g>
      </svg>
    </button>
    <button
      className="top-bar__btn top-bar__btn--expand"
      type="button"
      value="Expand"
    >
      <svg
        fill="currentColor"
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "color": undefined,
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 40 40"
        width="1em"
      >
        <g>
          <path
            d="m19.9 23.6q0 0.3-0.3 0.5l-7.4 7.4 3.2 3.2q0.5 0.4 0.5 1t-0.5 1-1 0.4h-10q-0.6 0-1-0.4t-0.4-1v-10q0-0.6 0.4-1t1-0.4 1 0.4l3.2 3.2 7.5-7.4q0.2-0.2 0.5-0.2t0.5 0.2l2.5 2.6q0.3 0.2 0.3 0.5z m17.4-19.3v10q0 0.6-0.4 1t-1 0.4-1-0.4l-3.3-3.2-7.4 7.4q-0.2 0.2-0.5 0.2t-0.5-0.2l-2.5-2.6q-0.3-0.2-0.3-0.5t0.3-0.5l7.4-7.4-3.2-3.2q-0.5-0.4-0.5-1t0.5-1 1-0.4h10q0.5 0 1 0.4t0.4 1z"
          />
        </g>
      </svg>
    </button>
  </div>
  <div
    className="d-inline-block center top-bar__right"
  >
    vatweb.fr
  </div>
</div>
`;
